src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -482,9 +482,18 @@ private BlittableJsonReaderObject RecreateCountersIfNeeded(DocumentsOperationCon
                 configuration.MinimumRevisionAgeToKeep.HasValue == false)
                 return;
 
-            var numberOfRevisionsToDelete = revisionsCount - configuration.MinimumRevisionsToKeep ?? 0;
-            if (numberOfRevisionsToDelete <= 0)
-                return;
+            long numberOfRevisionsToDelete;
+            if (configuration.MinimumRevisionsToKeep.HasValue)
+            {
+                numberOfRevisionsToDelete = revisionsCount - configuration.MinimumRevisionsToKeep.Value;
+                if (numberOfRevisionsToDelete <= 0)
+                    return;
+            }
+            else
+            {
+                // delete all revisions which age has passed
+                numberOfRevisionsToDelete = long.MaxValue;
```

Comment:
I have some doubts about that. If both of the settings `MinimumRevisionsToKeep` and `MinimumRevisionAgeToKeep` are `null` then we want to keep unlimited amount of revisions. What will happen now?

Reply:
If both are null it won't reach this point.

Reply:
Ok. Makes sense. I see the code now. The descriptions in the Studio match all our behaviors?

Reply:
Yes